UserSessionProviderTest#testOnClientRemoved fails on CockroachDB

### Area

storage

### Describe the bug

`UserSessionProviderTest#testOnClientRemoved` fails on CockroachDB with a NPE:

```
Caused by: java.lang.NullPointerException
	at org.keycloak.testsuite.model.UserSessionProviderTest.testOnClientRemoved(UserSessionProviderTest.java:340)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.keycloak.testsuite.rest.TestingResourceProvider.runModelTestOnServer(TestingResourceProvider.java:857)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
```

Test started failing after commit https://github.com/keycloak/keycloak/commit/f6985949b65da2dfb527e7127fad3d31f0dbba60 was introduced. This also relates to https://github.com/keycloak/keycloak/commit/f84fdfa8ef2071d4e7e8f48702234a1e604e55d4 that fixed several similar issues with the test. The fix for this new issue should probably follow the same pattern of using a separate transaction to access the created sessions on CRDB.

### Version

main

### Expected behavior

All tests should pass

### Actual behavior

testOnClientRemoved fails with NPE

### How to Reproduce?

Run the test with JPA Map Storage on CockroachDB

### Anything else?

_No response_

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java
@@ -328,32 +328,35 @@ public class UserSessionProviderTest extends AbstractTestRealmKeycloakTest {
     @Test
     @ModelTest
     public void testOnClientRemoved(KeycloakSession session) {
-        RealmModel realm = session.realms().getRealmByName("test");
         UserSessionModel[] sessions = createSessions(session);
 
-        String thirdPartyClientUUID = realm.getClientByClientId("third-party").getId();
-
-        Map<String, Set<String>> clientSessionsKept = new HashMap<>();
-        for (UserSessionModel s : sessions) {
-            // session associated with the model was closed, load it by id into a new session
-            s = session.sessions().getUserSession(realm, s.getId());
-            Set<String> clientUUIDS = new HashSet<>(s.getAuthenticatedClientSessions().keySet());
-            clientUUIDS.remove(thirdPartyClientUUID); // This client will be later removed, hence his clientSessions too
-            clientSessionsKept.put(s.getId(), clientUUIDS);
-        }
-
-        realm.removeClient(thirdPartyClientUUID);
-
         KeycloakModelUtils.runJobInTransaction(session.getKeycloakSessionFactory(), kcSession -> {
+            RealmModel realm = session.realms().getRealmByName("test");
+            String thirdPartyClientUUID = realm.getClientByClientId("third-party").getId();
+            Map<String, Set<String>> clientSessionsKept = new HashMap<>();
+
             for (UserSessionModel s : sessions) {
+                // session associated with the model was closed, load it by id into a new session
                 s = kcSession.sessions().getUserSession(realm, s.getId());
-                Set<String> clientUUIDS = s.getAuthenticatedClientSessions().keySet();
-                assertEquals(clientUUIDS, clientSessionsKept.get(s.getId()));
+                Set<String> clientUUIDS = new HashSet<>(s.getAuthenticatedClientSessions().keySet());
+                clientUUIDS.remove(thirdPartyClientUUID); // This client will be later removed, hence his clientSessions too
+                clientSessionsKept.put(s.getId(), clientUUIDS);
+            }
+
+            boolean clientRemoved = false;
+            try {
+                clientRemoved = realm.removeClient(thirdPartyClientUUID);
+
+                for (UserSessionModel s : sessions) {
+                    s = kcSession.sessions().getUserSession(realm, s.getId());
+                    Set<String> clientUUIDS = s.getAuthenticatedClientSessions().keySet();
+                    assertEquals(clientUUIDS, clientSessionsKept.get(s.getId()));
+                }
+            } finally {
+                // Revert client
+                if (clientRemoved) realm.addClient("third-party");
             }
         });
-
-        // Revert client
-        realm.addClient("third-party");
     }
 
     @Test

